core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -845,7 +847,7 @@ public synchronized void save() {
      * Loads the data from the disk into this object.
      */
     public synchronized void load() throws IOException {
-        UpdateSite defaultSite = new UpdateSite(ID_DEFAULT, config.getUpdateCenterUrl() + "update-center.json");
+        UpdateSite defaultSite = new UpdateSite(LEGACY_ID_DEFAULT, config.getUpdateCenterUrl() + "update-center.json");
         XmlFile file = getConfigFile();
```

Comment:
I would go for creating a `protected` method thats creates the `UpdateSite` with the default id (obtained from the system property) as an argument.

This allows subclasses to override the creation. 

The `LEGACY_ID_DEFAULT` would not be needed in this case.